src/FileUtils/AgGridUtils.js

```diff
@@ -115,17 +118,18 @@ const _validateFormat = (rows, hasHeader, cols, options) => {
   for (let rowIndex = startIndex; rowIndex < rows.length; rowIndex++) {
     const row = rows[rowIndex];
     while (row[row.length - 1] === undefined && row.length > knownColsCount) row.pop();
-    if (row.length !== knownColsCount || row.includes(undefined)) {
+
+    if (row.length < knownColsCount && !row.includes(undefined)) {
       _forgeColumnsCountError(row, rowIndex + 1, cols, errors);
-      continue;
     }
 
     for (let colIndex = 0; colIndex < knownColsCount; colIndex++) {
       const { type, colOptions, acceptsEmptyFields, field } = colMeta[colIndex];
       const value = row[colIndex];
       if (value === undefined) continue;
 
-      const validationResult = ValidationUtils.isValid(value, type, colOptions, acceptsEmptyFields);
+      const typeStr = Array.isArray(type) ? type[0] : type;
```

Comment:
The function `_getColTypeFromTypeArray` already does this, but it manages more edge cases and possible configurations :slightly_smiling_face: 
Since you changed how `colMeta` is now built from `colsData`, maybe we don't need this anymore? (I think we can now use `type` directly)